# Profitable App Profiles for the App Store and Google Play Markets

I work for a company that build free apps for an English audience and we derive benefits from the in-app ads revenues. The goal of the current project focuses on identifying the types of mobile apps that are most likely to succeed in the App Store and Google Play markets. 

In this regards, I'll work as a data analyst and analyze two datasets from Google and Apple Stores to find the pattern and help our developers make informed, data-driven decisions to uncover which kinds of apps are most likely to attract and retain a large user base.

# Opening and Exploring the Data

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

![img](https://s3.amazonaws.com/dq-content/350/py1m8_statista.png) Source: [Statista](https://www.statista.com/statistics/276623/number-of-apps-available-in-leading-app-stores/)

Collecting data for over four million apps requires a significant amount of time and money, so I'll try to analyze a sample of data instead. As to avoid spending resources with collecting new data ourselves, I should first try to see whether I can find any relevant existing data at no cost. Luckily, these are two data sets that seem suitable for our purpose:

* [Google Play data set](https://www.kaggle.com/datasets/lava18/google-play-store-apps) containing data about approximately ten thousand Android apps from Google Play and you can download the data set directly from this [link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).

* [App Store data set](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately seven thousand iOS apps from the App Store, and you can download the data set directly from this [link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv).

I'll start by opening the two data sets and then continue with exploring the data.

## About dataset

This dataset is about app usage in AppleStore, which contains more than 7000 Apple iOS mobile application details, and was extracted from the iTunes Search API at the Apple Inc website. 

You can find the original dataset from [here](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps)


| Column name | Description | 
| :---: | :----: |
| "id" | App ID       | 
| "track_name"   | App Name        |
| "size_bytes"   | Size (in Bytes)        |
| "currency"   | Currency Type        |
| "price"   | Price amount        |
| "rating_count_tot"   | User Rating counts (for all version)        |
| "rating_count_ver"   | User Rating counts (for current version)        |
| "user_rating"   | Average User Rating value (for all version)        |
| "user_rating_ver"   | Average User Rating value (for current version)        |
| "ver"   | Latest version code        |
| "cont_rating"   | Content Rating        |
| "prime_genre"   | Primary Genre        |
| "sup_devices.num"   | Number of supporting devices        |
| "ipadSc_urls.num"   | Number of screenshots showed for display        |
| "lang.num"   | Number of supported languages        |
| "vpp_lic"   | Vpp Device Based Licensing Enabled        |


This dataset is about app usage in Google Play, and was sourced from [here](https://www.kaggle.com/datasets/lava18/google-play-store-apps/data)
|Column name|Description|
| :---: | :----: |
| "App" | Application name       | 
| "Category"   | Category the app belongs to        |
| "Rating"   | Overall user rating of the app        |
| "Reviews"   | Number of user reviews for the app        |
| "Size"   | Size of the app        |
| "Installs"   | Number of user downloads/installs for the app        |
| "Type"   | Paid or Free        |
| "Price"   | Price of the app|
|"Content Rating"| Age group the app is targeted at - Children / Mature 21+ / Adult |
|"Genres"|An app can belong to multiple genres|

To make it easier to explore both datasets, I’ll define a reusable function called ```explore_data()```. This function will help display specific rows in a more readable format and optionally show the number of rows and columns in the dataset.

In [3]:
# Creating explore_data() to explore the data

def explore_data( dataset, start, end, rows_and_columns = False ):
    dataset_slice = dataset[ start : end ]    
    for row in dataset_slice:
        print( row )
        print( '\n' ) # adds a new (empty) line after each row for better visulization

    if rows_and_columns:
        print( 'Number of rows:' , len( dataset ) )
        print( 'Number of columns:' , len( dataset[ 0 ] ) )


# Import data of Google Play and of App Store

from csv import reader

    # The Google Play data set
opened_file = open( 'googleplaystore.csv' )
read_file = reader( opened_file )
android = list( read_file )
android_header = android[ 0 ]
android = android[ 1: ]

    # The App Store data set
opened_file = open( 'AppleStore.csv' )
read_file = reader( opened_file )
ios = list( read_file )
ios_header = ios[ 0 ]
ios = ios[ 1: ]

# Find the rows and columns for each data set

## For Google Play Store:

In [5]:
print( android_header )
print( '\n' )
explore_data( android, 0, 3, True )

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


We see that the Google Play data set has 10841 apps and 13 columns. With a quick glance, the columns that might be useful for the purpose of our analysis are ```'App'```, ```'Category'```, ```'Reviews'```, ```'Installs'```, ```'Type'```, ```'Price', and 'Genres'```.

## For Apple Store:

In [8]:
print( ios_header )
print( '\n' )
explore_data( ios, 0, 3, True )

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


We have 7197 iOS apps in this data set with 16 columns, and the columns that seem interesting are: ```'track_name'```, ```'currency', 'price'```, ```'rating_count_tot'```, ```'rating_count_ver'```, and ```'prime_genre'```. Not all column names are self-explanatory in this case, yet details about each column can be found in the data set [documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home).

# Cleaning Data
Including removing wrong and repeated apps, non-English apps, non-free apps.

### Deleting Wrong Data

The Google Play data set has a dedicated [discussion section](https://www.kaggle.com/datasets/lava18/google-play-store-apps/discussion?sort=hotness), and I can see that [one of the discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) outlines an error for row 10472. 

I printed this row and compare it against the header and another row that is correct.

In [12]:
print( android[ 10472 ] )  # incorrect row
print( '\n' )
print( android_header )    # header
print( '\n' )
print( android[ 0 ] )      # correct row

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']



The row 10472 corresponds to the app Life Made WI-Fi Touchscreen Photo Frame, and we can see that the rating is 19. This is clearly off because the maximum rating for a Google Play app is 5 (as mentioned in the [discussions section](https://www.kaggle.com/datasets/lava18/google-play-store-apps/discussion/66015), and this problem is caused by a missing value in the `'Category'` column).

As a consequence, I'll delete this row. 

In [14]:
print( len( android ) )
del android[ 10472 ]
print( len( android ) )

10841
10840


For Apple Store, which also has has a dedicated [discussion section](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps/discussion?sort=undefined), and I see no message regarding error for this dataset. Hence, no wrong data is deleted for Apple Store data. 

## Removing deplicated entry
### For Google Play Store:

In [17]:
duplicate_apps = [ ]
unique_apps = [ ]

for app in android:
    name = app[ 0 ]
    if name in unique_apps:
        duplicate_apps.append( name )
    else:
        unique_apps.append( name )
    
print( 'Number of duplicate apps:' , len( duplicate_apps ) )
print( '\n' )
print( 'Examples of duplicate apps:' , duplicate_apps[ :10 ] )

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


In total, there are 1,181 cases where an app occurs more than once.

For instance, Instagram has four entries:

In [19]:
for app in android:
    name = app[ 0 ]
    if name == 'Instagram' :
        print( app )

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


To ensure accurate analysis, I obviously want to avoid counting the same app more than once, namely that I need to remove duplicate entries and retain only one row per app.

While I could remove duplicates randomly, there's a more informed approach. Specifically, by looking at the rows for the Instagram app printed above, I noticed that the main difference between the duplicates lies in the fourth column, which represents the number of ```reviews```, which suggest that the data was collected at different times.

Hence, I can use this to build a criterion for keeping rows. In other words, I won't remove rows randomly, but rather I'll keep the rows that have the highest number of reviews as the higher the number of reviews, the more reliable the ratings.

So to do that, I will:

- Create a dictionary where each key is a unique app name, and the value is the highest number of reviews of that app
- Use the dictionary to create a new data set, which will have only one entry per app (and I only select the apps with the highest number of reviews)

In [21]:
reviews_max = { }
for app in android:
    name = app[ 0 ]
    n_reviews = float( app[ 3 ] )
    if name in reviews_max and reviews_max[ name ] < n_reviews:
        reviews_max[ name ] = n_reviews
    elif name not in reviews_max:
        reviews_max[ name ] = n_reviews

In a previous code cell, I found that there are 1,181 cases where an app occurs more than once, so the length of our dictionary (of unique apps) should be equal to the difference between the length of our data set and 1,181.

In [23]:
print( 'Expected length:' , len( android ) - 1181 )
print( 'Actual length:' , len( reviews_max ) )

Expected length: 9659
Actual length: 9659


Now, I'll use the reviews_max dictionary to remove the duplicatesby by keeping only the rows with the highest number of reviews for each app. 

In the code cell below:

- I start by initializing two empty lists, ```android_clean``` and ```already_added```.
- I then loop through the ```android``` data set, and for every iteration:
    - I isolate the name of the app and the number of reviews.
    - I add the current row (```app```) to the ```android_clean``` list, and the app name (```name```) to the ```already_added``` list if:
        - The number of reviews of the current app matches the number of reviews of that app as described in the ```reviews_max``` dictionary; and
        - The name of the app is not already in the ```already_added``` list. I need to add this supplementary condition to account for those cases where the highest number of reviews of a duplicate app is the same for more than one entry (for example, the Box app has three entries, and the number of reviews is the same). If I just check for ```reviews_max[name] == n_reviews```, I'll still end up with duplicate entries for some apps.

In [25]:
android_clean = [ ]
already_added = [ ]
for row in android:
    name = row[ 0 ]
    n_reviews = float( row[ 3 ] )
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append( row )
        already_added.append( name )

Now let's quickly explore the new data set, and confirm that the number of rows is 9,659.

In [27]:
explore_data( android_clean, 0, 3, True )

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


### For Apple Store:

Just like what I've done for Google Play Store, I'll do the same for Apple Store data. 

In [29]:
# Checking duplicated Apps
duplicate_apps = [ ]
unique_apps = [ ]

for app in ios:
    name = app[ 0 ]
    if name in unique_apps:
        duplicate_apps.append( name )
    else:
        unique_apps.append( name )
    
print( 'Number of duplicate apps:' , len( duplicate_apps ) )

Number of duplicate apps: 0


Here there is no duplicated apps in ios store. So I'll move to the next step of data cleaning.

## Removing non-English Apps

### Part 1

We use English for the apps we develop at our company, and thus I'd like to analyze only the apps that are designed for an English-speaking audience. However, by exploring the data long enough, I find that both datasets have apps with names that suggest they are not designed for an English-speaking audience.

Below are some examples from both data sets:

In [32]:
print( ios[ 813 ][ 1 ] )
print( ios[ 6731 ][ 1 ] )

print( android_clean[ 4412 ][ 0 ] )
print( android_clean[ 7940 ][ 0 ] )

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


As I leanred that the numbers corresponding to the characters commonly used in an English text are all in the range 0 to 127, according to the [ASCII](https://en.wikipedia.org/wiki/ASCII) (American Standard Code for Information Interchange) system. Hence, based on this number range, I build a function that detects whether a character belongs to the set of common English characters or not. If the number is equal to or less than 127, then the character belongs to the set of common English characters.

In other words, if an app name contains a character that is greater than 127, then it probably means that the app has a non-English name.

In [34]:
# function to check English or not
def Eng_check( app_name ):
    for character in app_name:
        if ord( character ) > 127:
            return False
    return True

print( Eng_check ( "instagram" ) )
print( Eng_check ( "爱奇艺" ) )

True
False


The function seems to work fine, but some English app names use emojis or other symbols (™, — (em dash), – (en dash), etc.) that fall outside of the ASCII range. Because of this, I'll remove useful apps if I still use the function in its current form.

In [36]:
print( Eng_check ( 'Docs To Go™ Free Office Suite' ) )
print( Eng_check ( 'Instachat 😜' ) )

print( ord( '™' ) )
print( ord( '😜' ) )

False
False
8482
128540


### Part 2
I order to minimize the impact of data loss, I'll only remove an app if its name has more than three non-ASCII characters, and to do this, I’ve updated the function from the previous step:

- The function takes a string (app name) as input.
- It counts how many characters fall outside the ASCII range (0–127).
- If the count is more than three, it returns ```False```, identifying the app name as non-English.
- Otherwise, it returns ```True```, meaning the name is considered English and will be kept.

In [38]:
def Eng_check( app_name ):
    Eng_letter = 0
    
    for character in app_name:
        if ord( character ) > 127:
            Eng_letter += 1
    
    if Eng_letter > 3:
        return False
    else:
        return True

# Example app_name check
print( Eng_check( 'Docs To Go™ Free Office Suite' ) )
print( Eng_check( 'Instachat 😜' ) )
print( Eng_check( '爱奇艺PPS -《欢乐颂2》电视剧热播' ) )

True
True
False


Here I used new function to filter out non-English apps from both datasets.

In [40]:
non_Eng_android = [ ]
Eng_android = [ ]
for row in android_clean:
    app_name = row [ 0 ]
    if Eng_check ( app_name ):
        Eng_android.append( row ) 

non_Eng_ios = [ ]
Eng_ios = [ ]
for row in ios:
    app_name = row [ 1 ]
    if Eng_check ( app_name ):
        Eng_ios.append( row )

explore_data( Eng_android, 0, 3, True )
print( '\n' )
explore_data( Eng_ios, 0, 3, True )
        

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

I can see that we're left with 9614 Android apps and 6183 iOS apps.

## Isolating the Free Apps

As I mentioned in the introduction, our company only build apps that are free to download and install, and our main source of revenue consists of in-app ads. However, our data sets contain both free and non-free apps, and thus I'll need to isolate only the free apps for our analysis. 

Below, I isolate the free apps for both our data sets.

In [43]:
final_android = [ ]
final_ios = [ ]

for row in Eng_android:
    app_price = row [ 7 ]
    if app_price == "0":
        final_android.append( row )

for row in Eng_ios:
    app_price = row [ 4 ]
    if app_price == "0.0":
        final_ios.append( row )

print ( len( final_android ) )
print ( len( final_ios ) )

8864
3222


We're left with 8864 Android apps and 3222 iOS apps, which should be enough for our analysis.

## Most Common Apps by Genre

### Part 1

Once agin, as mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we then develop it further.
3. If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

I'll begin the analysis by getting a sense of the most common genres for each market. For this, I'll build a frequency table for the ```prime_genre``` column of the App Store data set, and the ```Genres``` and ```Category``` columns of the Google Play data set.

### Part 2

I'll build two functions I can use to analyze the frequency tables:

- One function to generate frequency tables that show percentages
- Another function that we can use to display the percentages in a descending order

In [46]:
def freq_table( dataset, index ):
    table = { }
    total = 0
    
    for row in dataset:
        total += 1
        value = row[ index ]
        if value in table:
            table[ value ] += 1
        else:
            table[ value ] = 1
    
    table_percentages = { }
    for key in table:
        percentage = ( table[ key ] / total) * 100
        table_percentages[ key ] = percentage 
    
    return table_percentages

def display_table( dataset, index ):
    table = freq_table( dataset, index )
    table_display = [ ]
    for key in table:
        key_val_as_tuple = ( table[ key ], key )
        table_display.append( key_val_as_tuple )

    table_sorted = sorted( table_display, reverse = True )
    for entry in table_sorted:
        print( entry[ 1 ], ':', entry[ 0 ] )

### Part 3

I start by examining the frequency table for the ```prime_genre``` column of the App Store data set.

In [48]:
display_table( final_ios, -5 )

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


I notice that among the free English apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed by social networking apps which amount for 3.29% of the apps in our data set.

The general impression is that App Store (at least the part containing free English apps) is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare. However, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer.

I continue by examining the ```Genres``` and ```Category``` columns of the Google Play data set (two columns which seem to be related).

In [50]:
display_table( final_android, 1 ) # Category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The landscape seems significantly different on Google Play: there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, by investigating this further, I notice that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

![Screenshot Image](https://s3.amazonaws.com/dq-content/350/py1m8_family.png)

Even so, practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the Genres column:

In [52]:
display_table( final_android, -4 )

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The difference between the ```Genres``` and the ```Category``` columns is not crystal clear, but one thing I notice is that the ```Genres``` column is much more granular (it has more categories). Since I am only looking for the bigger picture at the moment, I'll only work with the ```Category``` column moving forward.

Up to this point, I found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. 

Now I'd like to get an idea about the kind of apps that have most users.

## Most Popular Apps by Genre on the App Store

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, I find this information in the ```Installs``` column, but for the App Store data set this information is missing. As a workaround, I'll take the total number of user ratings as a proxy, which I can find in the ```rating_count_tot app```.

Below, I calculate the average number of user ratings per app genre on the App Store:

In [55]:
ios_genre = freq_table ( final_ios, -5 )
for genre in ios_genre:
    total = 0     # This variable will store the sum of user ratings specific to each genre
    len_genre = 0 # This variable will store the number of apps specific to each genre
    for app in final_ios:
        genre_app = app[ -5 ]
        if genre_app == genre:
            rating_app = float( app[ 5 ] )
            total += rating_app
            len_genre += 1
    avg_rating = total / len_genre
    print ( genre, ":", avg_rating ) 

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [57]:
for app in final_ios:
    if app[ -5 ] == 'Navigation':
        print( app[ 1 ], ':', app[ 5 ] ) # print name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.

Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

Reference apps have 74,942 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:

In [59]:
for app in final_ios:
    if app[ -5 ] == 'Reference':
        print(app[ 1 ], ':', app[ 5 ])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps, which suggests the market might be a bit saturated with for-fun apps, meaning that a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

- Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

- Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

- Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now I'll analyze the Google Play market.

## Most Popular Apps by Genre on Google Play

For the Google Play market, since I have data about the number of installs, it should be able to get a clearer picture about genre popularity. 

However, the install numbers don't seem precise enough — that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [62]:
display_table( final_android, 5 ) # the Installs columns

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


One problem with this data is that is not precise. For instance, I don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

In other wrods, we're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

To perform computations, however, I'll convert each install number to ```float``` — this means that I need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error. I'll do this directly in the loop below, where I also compute the average number of installs for each genre (category).

In [64]:
category_android = freq_table( final_android, 1 )
for category in category_android:
    total = 0
    len_category = 0
    for app in final_android:
        category_app = app[ 1 ]
        if category_app == category:            
            n_installs = app[ 5 ]
            n_installs = n_installs.replace( ',', '' )
            n_installs = n_installs.replace( '+', '' )
            total += float( n_installs )
            len_category += 1
    avg_install = total / len_category 
    print( category, ':', avg_install )

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

On average, communication apps have the most installs: 38,456,119, which is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [66]:
for app in final_android:
    if app[ 1 ] == 'COMMUNICATION' and ( app[ 5 ] == '1,000,000,000+'
                                      or app[ 5 ] == '500,000,000+'
                                      or app[ 5 ] == '100,000,000+' ):
        print( app[ 0 ], ':', app[ 5 ] )

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

If I removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [68]:
under_100_m = [ ]

for app in final_android:
    n_installs = app[ 5 ]
    n_installs = n_installs.replace( ',', '' )
    n_installs = n_installs.replace( '+', '' )
    if ( app[ 1 ] == 'COMMUNICATION' ) and ( float( n_installs ) < 100000000 ):
        under_100_m.append( float( n_installs ) )
        
sum( under_100_m ) / len( under_100_m )

3603485.3884615386

I see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously I found out this part of the market seems a bit saturated, so I'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

I'll take a look at some of the apps from this genre and their number of installs:

In [70]:
for app in final_android:
    if app[ 1 ] == 'BOOKS_AND_REFERENCE' :
        print( app[ 0 ], ':', app[ 5 ] )

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [72]:
for app in final_android:
    if app[ 1 ] == 'BOOKS_AND_REFERENCE' and ( app[ 5 ] == '1,000,000,000+'
                                            or app[ 5 ] == '500,000,000+'
                                            or app[ 5 ] == '100,000,000+' ):
        print( app[ 0 ], ':', app[ 5 ] )

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential. I'll try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [74]:
for app in final_android:
    if app[ 1 ] == 'BOOKS_AND_REFERENCE' and ( app[ 5 ] == '1,000,000+'
                                            or app[ 5 ] == '5,000,000+'
                                            or app[ 5 ] == '10,000,000+'
                                            or app[ 5 ] == '50,000,000+' ):
        print( app[ 0 ], ':', app[ 5 ] )

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

This niche appears to be dominated by apps focused on processing and reading ebooks, along with various collections of libraries and dictionaries. Given the high level of competition, developing a similar app might not be the best strategy.

However, I also noticed a number of apps centered around the Quran, which suggests that creating apps based on popular books can be profitable. This points to a potential opportunity: transforming a well-known or recent book into an engaging app could perform well in both the Google Play and App Store markets.

That said, since the market is already saturated with basic library-style apps, we’d need to differentiate our product by adding unique features—such as daily quotes, audio versions, interactive quizzes, or even a discussion forum for readers.

## Conclusions

In this project, I analyzed data from the App Store and Google Play to identify an app profile that could be profitable in both markets.

Based on the analysis, one promising idea is to develop an app centered around a popular book—ideally a recent or trending title. While the market is already saturated with basic library and ebook apps, there's potential for success if the app offers additional features beyond just the text. These could include daily quotes, an audio version, interactive quizzes, or a community forum for discussion. Such enhancements could help the app stand out and attract a larger, more engaged user base.